osidb/tests/test_flaw.py

```diff
@@ -791,3 +798,13 @@ def test_validate_future_unembargo_date(self):
         with freeze_time(future_dt):
             FlawFactory(unembargo_dt=future_dt, embargoed=True)
             # no exception should be raised now
+
+    def test_validate_cvss3(self):
+        """
+        Test that the ValidationError is not raised when the flaw has a CVSS3 string
+        """
+        assert FlawFactory(cvss3="3.7/CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:L/A:N")
+
+        with pytest.raises(ValidationError) as e:
```

Comment:
nitpick: the test case description does not mention this one :smile: I would maybe just split it into two different tests so when either flaw with or without CVSS string fails the test it is immediately clear which case failed

Reply:
makes sense, would a parameterized approach like this works or do you think it would be better two completely separated tests?
```
    @pytest.mark.parametrize(
        "cvss3,should_raise",
        [
            ("3.7/CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:L/A:N", False),
            (None, True),
        ],
    )
    def test_validate_cvss3(self,cvss3,should_raise):
        """
        Test that the ValidationError is not raised when the flaw has a CVSS3 string
        """
        if should_raise:
            with pytest.raises(ValidationError) as e:
                FlawFactory(cvss3=cvss3)
            assert "CVSSv3 score is missing" in str(e)
        else:
            assert FlawFactory(cvss3=cvss3)
```

Reply:
I have no strong preference here so completely up to you :wink: